customcalendarview/src/main/java/com/stacktips/view/CustomCalendarView.java

```diff
@@ -0,0 +1,1010 @@
+                    break;
+                case 3 :
+                    helperInitializeWeekLayout(weekDaysArray, i, ResourceTable.Id_dayOfWeek3);
+                    break;
+                case 4 :
+                    helperInitializeWeekLayout(weekDaysArray, i, ResourceTable.Id_dayOfWeek4);
+                    break;
+                case 5 :
+                    helperInitializeWeekLayout(weekDaysArray, i, ResourceTable.Id_dayOfWeek5);
+                    break;
+                case 6 :
+                    helperInitializeWeekLayout(weekDaysArray, i, ResourceTable.Id_dayOfWeek6);
+                    break;
+                case 7 :
+                    helperInitializeWeekLayout(weekDaysArray, i, ResourceTable.Id_dayOfWeek7);
+                    break;
+                default :
+                    break;
+            }
+        }
+    }
+
+
+    private void helperSetDaysInCalendar(Calendar calendar, Calendar startCalendar, int i, int monthEndIndex,
+                                         int[] dayOfMonthIds) {
+        ComponentContainer dayOfMonthContainer = (ComponentContainer) mView.findComponentById(dayOfMonthIds[0]);
+        DayView dayView = (DayView) mView.findComponentById(dayOfMonthIds[1]);
+        if (dayView == null) {
+            return;
+        }
+        //Apply the default styles
+        dayOfMonthContainer.setClickedListener(null);
+        dayView.bind(startCalendar.getTime(), getDecorators());
+        dayView.setVisibility(Component.VISIBLE);
+        dayOfMonthContainer.setTag(DAY_OF_MONTH_CONTAINER + i);
+        if (null != getCustomTypeface()) {
+            dayView.setFont(getCustomTypeface());
+        }
+        if (CalendarUtils.isSameMonth(calendar, startCalendar)) {
+            dayOfMonthContainer.setClickedListener(onDayOfMonthClickListener);
+            ShapeElement shapeElement = new ShapeElement();
+            shapeElement.setShape(ShapeElement.RECTANGLE);
+            shapeElement.setRgbColor(RgbColor.fromArgbInt(calendarBackgroundColor.getValue()));
+            dayView.setBackground(shapeElement);
+            dayView.setTextColor(dayOfWeekTextColor);
+            //Set the current day color
+            markDayAsCurrentDay(startCalendar);
+        } else {
+            ShapeElement shapeElement = new ShapeElement();
+            shapeElement.setShape(ShapeElement.RECTANGLE);
+            shapeElement.setRgbColor(RgbColor.fromArgbInt(disabledDayBackgroundColor.getValue()));
+            dayView.setBackground(shapeElement);
+            dayView.setTextColor(disabledDayTextColor);
+            if (!isOverflowDateVisible() || (i >= 36 && (monthEndIndex / 7.0f) >= 1)) {
+                dayView.setVisibility(Component.HIDE);
+            }
+        }
+        dayView.decorate();
+        startCalendar.add(Calendar.DATE, 1);
+    }
+
+    private void switchSetDaysInCalendar2(Calendar calendar, Calendar startCalendar, int i, int monthEndIndex) {
```

Comment:
1. Place the extended functions of the switch case in order
2.  Rename the extended functions as switchSetDaysInCalendarExtended1 & switchSetDaysInCalendarExtended2